src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/jvm/sizing/HeapSizeIncreasePolicy.java

```diff
@@ -84,71 +87,41 @@ private void addToSlidingWindow() {
     List<HotNodeSummary> hotNodeSummaries = flowUnit.getSummary().getHotNodeSummaryList();
     hotNodeSummaries.forEach(hotNodeSummary -> {
       NodeKey nodeKey = new NodeKey(hotNodeSummary.getNodeID(), hotNodeSummary.getHostAddress());
-      perNodeSlidingWindow.next(nodeKey, new SlidingWindowData(currTime, 1d));
+      heapSizeIncreaseClusterMonitor.recordIssue(nodeKey, currTime);
     });
   }
 
-  private List<Action> evaluateAndEmit() {
-    List<Action> actions = new ArrayList<>();
-    int numNodesInCluster = appContext.getAllClusterInstances().size();
-    int numNodesInClusterUndersizedOldGen = getUnderSizedOldGenCount();
+  private class HeapSizeIncreaseClusterMonitor {
 
-    if ((numNodesInClusterUndersizedOldGen / (double) numNodesInCluster) * 100d >= unhealthyNodePercentage) {
-      Action jvmSizeUpAction = new HeapSizeIncreaseAction(appContext);
-      if (jvmSizeUpAction.isActionable()) {
-        actions.add(jvmSizeUpAction);
-      }
-    }
+    private static final int DEFAULT_DAY_BREACH_THRESHOLD = 8;
+    private static final int DEFAULT_WEEK_BREACH_THRESHOLD = 3;
+    private final Map<NodeKey, AlarmMonitor> perNodeMonitor;
```

Comment:
Good enough for L0, can you add a TODO comment here, I think we can reduce the amount of memory this consumes by making a custom SlidingWindow for L1

Reply:
I thought so too. It's already a 1000 line PR, didn't want to add to it, and also cover it in testing for this time.